File: Backend/Monolith.DAL/Models/Product.cs
using Monolith.DAL.Contracts;
using System;
using System.Collections.Generic;

namespace Monolith.DAL.Models
{
   public class Product : ITrackable
   {
      public int Id { get; set; }

      public string Name { get; set; }
      public string Description { get; set; }
      public string ImageUrl { get; set; }
      public double Price { get; set; }
      public bool DelFlag { get; set; }
      public int CategoryId { get; set; }
      public Category Category { get; set; }

      public virtual ICollection<OrderProduct> OrderProducts { get; set; }

      public DateTime CreatedAt { get; set; }
      public DateTime ModifiedAt { get; set; }
   }
}

File: Frontend/Monolith-SPA/src/app/_models/Product.ts
import { Category } from './Category';
import { Size } from './Size';

export class Product {
    id?: number;
    name: string;
    description: string;
    imageUrl: string;
    price: number;
    categoryId: number;
    category: Category;
    size: Size;
    createdAt?: Date;
    modifiedAt?: Date;

    get totalPrice(): number {
        let sum: number;
        if (this.size != null) {
            sum = this.price + this.size.costMultiplier;
        }
        sum = this.price;
        return sum;
    }
}
